Build: #BBB-92 서적 관련 app, domain, infra, support 모듈 분리

platinouss opened the pull request:
## 작업 개요
<!-- 작업에 대한 요약 설명을 남겨주세요. -->
멀티 모듈 단일 프로젝트를 적용하기 위해, 서적과 관련된 app, domain, infra, support 모듈을 분리하여 적용했습니다.

## 전달 사항
<!-- 작업과 관련된 전달 사항을 남겨주세요. --> 
- **조금 더 조사해보고 domain 모듈에 존재하는 dto 객체를 적절한 위치와 타입으로 수정하겠습니다.**
> 실제로 domain 계층에서는 vo(value object) 정도만 존재하는게 맞는 것 같지만, 실제 vo의 역할이 repository에 인자로 넘기는 역할이 아닌 것 같아서 조금 더 고민이 필요해보입니다. (클린 아키텍처의 유스케이스 개념과 관련이 있을지?)

### 추가적으로 알아보아야 할 것
- 현재 repository에서 공통적으로 사용 할 순수 객체를 domain 모듈에 선언했습니다. 만약 해당 객체의 일부 필드만 사용하여 데이터 액세스 계층에 접근해야 하는 경우, 해당 필드만 가지고 있는 dto? vo? 객체를 선언해야 할 것입니다. 이 상항에서 해당 객체를 어느 모듈에 선언하고 어떻게 선언해야 할까요 ?
- 서적 도메인에서는 MySQL과 Elasticsearch 저장소에 모두 접근해야 하는 상황인데, 현재 도메인 계층에 선언된 하나의 repository 인터페이스에 두 저장소에 접근하는 메서드를 모두 추가해둔 상태입니다 (도메인 계층에서는 데이터 액세스 계층에 의존하지 않도록). 하지만 메서드가 많아지면 해당 reposiotry 인터페이스에 모두 몰리게 되어 최선의 구조는 아닌 것 같은데 어떤 식으로 구성해야 최선일지?
- CQRS 패턴이 적용된 command와 query 객체는 어떤 레이어에서 해체가 되어야 할까?

## 참고 자료
<!-- 작업 시 참고했던 자료의 출처를 남겨주세요. -->

platinouss committed:
⚙️Build: 서적 관련 app, domain, infra, support 모듈 분리

platinouss committed:
⚙️Build: 모듈 import 경로 수정 및 불필요한 파일 제거
[BBB-92]
아직 멀티 모듈이 적용되지 않은 study 모듈에서 필요로 하는 의존성을 임시로 추가하고, 중복으로 존재하는 test config를 제거했습니다.

platinouss committed:
🛠️Refactor: BookReader & BookWriter 제거

sseunghoon commented on support/settings.gradle:
```diff
@@ -0,0 +1,3 @@
+rootProject.name = 'devs'
+
+include ':support:common', ':support:naver-client'
```
서브 모듈 속에서도 root 프로젝트를 만들 수 있는게 신기하네요

sseunghoon reviewed (approved):
Repository 를 순수 인터페이스로 두고
JPARepository 들을 JPARepository라고 작명한거 좋은거 같아요

고생하셨습니다 👍

msjang4 commented on app/external-api/src/main/java/com/bombombom/devs/external/book/service/dto/SearchBooksResult.java:
```diff
@@ -0,0 +1,65 @@
+            .build();
+    }
+
+    public static SearchBooksResult fromNaverBookApiResult(NaverBookApiResult naverBookApiResult) {
+        List<BookResult> bookResults = naverBookApiResult.items().stream()
+            .filter(item -> Objects.nonNull(item.isbn()))
+            .map(item -> BookResult.builder()
+                .title(item.title())
+                .author(item.author())
+                .publisher(item.publisher())
+                .isbn(item.isbn())
+                .imageUrl(item.image())
+                .build()
+            ).toList();
+        return SearchBooksResult.builder().bookResults(bookResults).build();
+    }
```
좋네요 서비스 단에서 반환하는 Dto를 SearchBooksResult로 통일하면 컨트롤러에서 작업하기 좋은것 같습니다

msjang4 reviewed (approved):
고생하셨습니다 굿

platinouss committed:
⚡️Fix: 테스트 진행 시 기존에 도커환경에서 실행되던 Elasticsearch가 죽는 이슈 해결
[BBB-92]
테스트 전용 ES 컨테이너를 올렸는데, 기존에 실행되고 있던 ES가 137 에러 코드를 던지고 죽는 문제 발생. 테스트 시 도커에 올라가는 ES의 힙 공간을 최소한으로 제한하여 해결했습니다.

platinouss committed:
🛠️Refactor: 서적을 색인할 때 vo 객체를 활용하도록 수정

platinouss committed:
⚡️Fix: ES에 저장된 해당 서적의 bookId가 업데이트되지 않는 문제 해결
[BBB-92]
특정 서적이 기술 서적으로 선정되어 DB에 저장되면, Elasticsearch에서 해당 서적의 book id가 업데이트 되지 않는 문제를 해결했습니다.
